src/models/userFamily.ts

```diff
@@ -0,0 +1,56 @@
+import type { PopulatedDoc, Types, Document, Model } from "mongoose";
+import { Schema, model, models } from "mongoose";
+import type { InterfaceUser } from "./User";
+/**
+ * This is an interface that represents a database(MongoDB) document for Family.
+ */
+
+export interface InterfaceUserFamily {
+  _id: Types.ObjectId;
+  title: string;
+  users: PopulatedDoc<InterfaceUser & Document>[];
+  admins: PopulatedDoc<InterfaceUser & Document>[];
+  creator: PopulatedDoc<InterfaceUser & Document>[];
+}
+
+/**
+ * @param  title - Name of the user Family (type: String)
+ * Description: Name of the user Family.
+ */
+
+/**
+ * @param  users - Members associated with the user Family (type: String)
+ * Description: Members associated with the user Family.
+ */
+const UserFamilySchema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  users: [
+    {
+      type: Schema.Types.ObjectId,
+      ref: "User",
+      required: true,
+    },
+  ],
+  admins: [
+    {
+      type: Schema.Types.ObjectId,
+      ref: "User",
+      required: true,
+    },
+  ],
+  creator: {
+    type: Schema.Types.ObjectId,
+    ref: "User",
+    required: true,
+  },
```

Comment:
This data is repeated and adds complexity to the resolvers to maintain integrity. Either store them here or on the user object. According to me these all are better stored here, but Im open to counter arguments.

Reply:
Committed all the asked changes the PR is up for review now. Thanks for suggestions.